### ISSUE TITLE
Support for ECSCredentials

### ISSUE THREAD

**Issue by Larisho** [cloudydeno/deno-aws_api#33]
We're trying to use this package in a deno app deployed via ECS and we're getting credential errors when trying to connect to dynamodb. On inspection of `lib/client/credentials.ts`, I noticed that ECSCredential handling is commented out/not included.

https://github.com/cloudydeno/deno-aws_api/blob/a383db0f0793f0211be46f039ccce9b3e49f8974/lib/client/credentials.ts#L55

 Adding support for this would be really helpful.

**Comment by danopia** [comment]
Hello, thank you for the request! I haven't had an ECS environment handy, which meant I couldn't have verified the code if I had written it already.

I'll put together an implementation and respond here so you could ideally verify it in your environment. That would help me be confident about merging it into main and shipping a new minor version.

**Comment by Larisho** [comment]
That would be great. Really appreciate it

**Comment by danopia** [comment]
Hi again, not sure how convenient it is for you to verify a raw version from an open PR. I made [a small `get-ecs-identity.ts` example](https://gist.github.com/danopia/36788750e91fb721ad4d61bc33a61e2c) which can be run directly:

```
deno run --no-prompt --allow-env --allow-net https://gist.githubusercontent.com/danopia/36788750e91fb721ad4d61bc33a61e2c/raw/a4a3adba8ad512055bdda05b2e9fd48190e05e95/get-ecs-identity.ts
```

I did some sanity checks locally but I haven't seen a full example response from ECS so there's a little bit of guessing. Let me know if you aren't able to verify and I could try briefly running ECS in my own account instead.
